packages/astro/src/runtime/server/index.ts

```diff
@@ -519,6 +527,10 @@ Make sure to use the static attribute syntax (\`${key}={value}\`) instead of the
 		return markHTMLString(` ${key.slice(0, -5)}="${toAttributeString(serializeListValue(value))}"`);
 	}
 
+	if (key === 'style' && typeof value === 'object') {
+		return markHTMLString(` ${key}="${toStyleString(value)}"`);
+	}
```

Comment:
OK sneaking a small feature in here, `style={{ color: "red" }}` support. Happy to remove if there's any opposition.

Reply:
Hate to be that reviewer, but I'd say pull this one out for now.  It's an interesting quick feature for a separate PR though!